### PR TITLE
Collision components rework

### PR DESCRIPTION
**Pull request by Ceanze**
- Collision components no longer uses any inherit from a "CollisionComponent".
- GameLogic no longer is a pass though for collision handling.
- All collision components currently listen to the PlayerCollisionEvent, but before specific code is executed, a check to see if it affects them is done.

### CODE REVIEW

File: GameProject/Engine/Components/PlayerCollision.cpp [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
 
-PlayerCollision::PlayerCollision(Entity* parentEntity, const std::string& tagName) : CollisionComponent(parentEntity, tagName)
+PlayerCollision::PlayerCollision(Entity* parentEntity, const std::string& tagName) : Component(parentEntity, tagName)
 {
+	EventBus::get().subscribe(this, &PlayerCollision::collide);
 }
 
 
 PlayerCollision::~PlayerCollision()
 {
```
Comment (TheoBerlin) [inline GameProject/Engine/Components/PlayerCollision.cpp]: Unsubscribe from collision events when deleting the component.

File: GameProject/Engine/Components/TargetMovingCollision.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#include "../Events/Events.h"
+#include "Component.h"
+
+/*
+	This class is added to all of the targets (drones and whatnot) that should react when a collision to them has happened.
+*/
+
+namespace reactphysics3d { class ProxyShapes; }
+
+class TargetMovingCollision : public Component
```
Comment (TheoBerlin) [inline GameProject/Engine/Components/TargetMovingCollision.h]:
```suggestion
class MovingTargetCollision : public Component
```

File: GameProject/Engine/Components/TargetStaticCollision.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#include "../Events/EventBus.h"
+#include "Component.h"
+
+/*
+	This class is added to all of the targets (drones and whatnot) that should react when a collision to them has happened.
+*/
+
+namespace reactphysics3d { class ProxyShapes; }
+
+class TargetStaticCollision : public Component
```
Comment (TheoBerlin) [inline GameProject/Engine/Components/TargetStaticCollision.h]:
```suggestion
class StaticTargetCollision : public Component
```

File: GameProject/Engine/Components/PlayerCollision.h [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
+#include "Component.h"
+#include "Engine/Events/Events.h"
 
-class PlayerCollision : public CollisionComponent
+namespace reactphysics3d { class ProxyShape; }
+
+class PlayerCollision : public Component
 {
 public:
 	PlayerCollision(Entity* parentEntity, const std::string& tagName = "Collision");
```
Comment (TheoBerlin) [inline GameProject/Engine/Components/PlayerCollision.h]:
```suggestion
	PlayerCollision(Entity* parentEntity, const std::string& tagName = "PlayerCollision");
```

File: GameProject/Engine/Components/TargetMovingCollision.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+{
+public:
+	TargetMovingCollision(Entity * parentEntity, const std::string& tagName = "Collision");
```
Comment (TheoBerlin) [inline GameProject/Engine/Components/TargetMovingCollision.h]:
```suggestion
	TargetMovingCollision(Entity * parentEntity, const std::string& tagName = "MovingTargetCollision");
```

File: GameProject/Engine/Components/TargetStaticCollision.h [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+{
+public:
+	TargetStaticCollision(Entity * parentEntity, const std::string& tagName = "Collision");
```
Comment (TheoBerlin) [inline GameProject/Engine/Components/TargetStaticCollision.h]:
```suggestion
	TargetStaticCollision(Entity * parentEntity, const std::string& tagName = "StaticTargetCollision");
```

File: GameProject/Game/Level/LevelParser.cpp [hunk 7]
```diff
@@ -43,13 +44,14 @@ void LevelParser::readEntityTargets(Level& level)
 		if (!path.empty()) {
 			// The target is mobile
 			level.targetManager->addMovingTarget(entity, path);
+			new TargetMovingCollision(entity);
```
Comment (TheoBerlin) [inline GameProject/Game/Level/LevelParser.cpp]: Move this and line 51 to `TargetManager::addMovingTarget` and `TargetManager::addStaticTarget` respectively.

File: GameProject/Engine/Components/PlayerCollision.cpp (hunk 1 above)
Comment (Ceanze) [reply GameProject/Engine/Components/PlayerCollision.cpp]: Added

File: GameProject/Engine/Components/TargetMovingCollision.h (hunk 2 above)
Comment (Ceanze) [reply GameProject/Engine/Components/TargetMovingCollision.h]: Renamed

File: GameProject/Engine/Components/TargetStaticCollision.h (hunk 3 above)
Comment (Ceanze) [reply GameProject/Engine/Components/TargetStaticCollision.h]: Renamed

File: GameProject/Engine/Components/PlayerCollision.h (hunk 4 above)
Comment (Ceanze) [reply GameProject/Engine/Components/PlayerCollision.h]: Renamed

File: GameProject/Engine/Components/TargetMovingCollision.h (hunk 5 above)
Comment (Ceanze) [reply GameProject/Engine/Components/TargetMovingCollision.h]: Renamed

File: GameProject/Engine/Components/TargetStaticCollision.h (hunk 6 above)
Comment (Ceanze) [reply GameProject/Engine/Components/TargetStaticCollision.h]: Renamed

File: GameProject/Game/Level/LevelParser.cpp (hunk 7 above)
Comment (Ceanze) [reply GameProject/Game/Level/LevelParser.cpp]: Moved to those

**Pull request merged**